src/wallet/rpcwallet.cpp

```diff
@@ -111,7 +111,7 @@ bool EnsureWalletIsAvailable(const CWallet* pwallet, bool avoidException)
         "Wallet file not specified (must request wallet RPC through /wallet/<filename> uri-path).");
 }
 
-void EnsureWalletIsUnlocked(const CWallet* pwallet)
+void EnsureWalletIsUnlocked(const std::shared_ptr<CWallet>& pwallet)
```

Comment:
This is also called when the pointer is never `nullptr`, so a reference is more appropriate.
```suggestion
void EnsureWalletIsUnlocked(const CWallet& wallet)
```

Reply:
Adapted. Unit tests passing.

Reply:
Yeah, my bad.